### DISCUSSION TITLE
How to return table data from custom Function

### DISCUSSION THREAD

**Discussion by taylorhlabs** [supabase/supabase#3421] [Questions] [answered]
I am trying to write a custom function in the Database > Functions editor that returns table data based on a custom query (confirmed working as a standalone SELECT statement) but the editor keeps giving me syntax errors. It seems that the CREATE FUNCTION statement can be left out when using the Function Editor but I'm not 100% sure. Could someone provide an example of exactly what code can be entered into the Function Editor to return table data from a simple SELECT statement (ie SELECT * FROM some_table)? Thanks!

**Comment by dshukertjr** [comment, marked as answer]
Hi @taylorhlabs!

I was able to create a new function using SQL by changing the language under advanced setting to SQL. I think I forgot the `;` at the end of my SQL statement in the screenshot, but other than that, it should work! Hopefully this helps!

<img width="670" alt="Screen Shot 2021-10-06 at 8 24 53" src="https://user-images.githubusercontent.com/18113850/136116243-d89c0f63-ff25-489d-b2f5-fa54bae13c35.png">
